web/src/main/webapp/WEB-INF/data/config/schema_plugins/iso19139/index-fields.xsl

```diff
@@ -255,19 +255,33 @@
 
 				<!-- Index platforms -->
 
-				<xsl:if test="gmd:thesaurusName/*/gmd:title/*/text()='AODN Platform Vocabulary'">
+				<xsl:if test="gmd:thesaurusName/*/gmd:title/*/text()='theme.aodn_aodn-platform-vocabulary.rdf'">
 					<xsl:for-each select="gmd:keyword/*">
 						<Field name="platform" string="{text()}" store="true" index="true"/>
-						<Field name="platformUri" string="{@xlink:href}" store="true" index="true"/>
+						<xsl:choose>
+							<xsl:when test="@xlink:href">
+								<Field name="platformKeyword" string="{@xlink:href}" store="true" index="true"/>
+							</xsl:when>
+							<xsl:otherwise>
+								<Field name="platformKeyword" string="{text()}" store="true" index="true"/>
+							</xsl:otherwise>
+						</xsl:choose>
 					</xsl:for-each>
 				</xsl:if>
 
 				<!-- Index parameters -->
 
-				<xsl:if test="gmd:thesaurusName/*/gmd:title/*/text()='AODN Discovery Parameter Vocabulary'">
```

Comment:
As discussed,  we need to keep this - GeoNetwork 3 and the the GA transform will put the correct Thesaurus title in  the metadata i.e. 'AODN Discovery Parameter Vocabulary'.

Perhaps we should make the changes Sachit made in GeoNetwork 3 so we can test this easily 

See https://github.com/geonetwork/core-geonetwork/pull/4296